protocol/directory.go

```diff
@@ -117,6 +189,25 @@ func (d *ConiksDirectory) KeyLookup(req *KeyLookupRequest) (
 	return NewKeyLookupProof(ap, d.LatestSTR(), nil, ErrorNameNotFound)
 }
 
+// KeyLookupInEpoch gets the public key for the username for a prior
+// epoch in the directory history indicated in the
+// KeyLookupInEpochRequest req received from a CONIKS client,
+// and returns a tuple of the form (response, error).
+// The response (which also includes the error code) is supposed to
+// be sent back to the client. The returned error is used by the key
+// server for logging purposes.
+//
+// A request without a username or with an epoch greater than the latest
+// epoch of this directory is considered malformed, and causes
+// KeyLookupInEpoch() to return a (error response,
+// ErrorMalformedClientMessage) tuple.
+// If the username doesn't have an entry in the directory,
+// at the indicated snapshot, KeyLookupInEpoch() returns a (KeyLookupInEpoch
```

Comment:
"Epoch" works, though, I was trying to use PAD-terminology here, since `KeyLookupInEpoch()` is really looking in the snapshot for the indicated epoch. I'll clarify this.